src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1328,6 +1330,65 @@ class FunctionBinaryArithmetic : public IFunction
         return function->execute(new_arguments, result_type, input_rows_count);
     }
 
+    ColumnPtr executeComparsion(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const
+    {
+        bool args_have_nulls = false;
```

Comment:
The code in l. 1335 - 1390 is probably copypasta from elsewhere in this file. Kindly add a comment that explains the reader _why_ a special case was implemented.